File: Java Back End/src/InfoSecCooker/Communication/Messages/CommandMessages/GraphBuilding/AddSinkTaskCommandMsg.java
package InfoSecCooker.Communication.Messages.CommandMessages.GraphBuilding;

public class AddSinkTaskCommandMsg extends AddSourceSinkTaskCommandMsg
{
    public AddSinkTaskCommandMsg(String taskName, long taskTemporaryId, String additionalInfo)
    {
        super(taskName, taskTemporaryId, additionalInfo);
    }
}

File: JavaScript Front End/src/Messages/CommandMessages/GraphBuilding/AddSinkTaskCommandMsg.js
const AddSourceSinkTaskCommandMessage = require('./AddSourceSinkTaskCommandMsg.js').AddSourceSinkTaskCommandMessage;

class AddSinkTaskCommandMsg extends AddSourceSinkTaskCommandMessage
{
    constructor(taskName, taskTemporaryId, additionalInfo)
    {
        super(taskName, taskTemporaryId, additionalInfo);
        this.jsonMessage["@type"] = this.jsonMessage["@type"].concat("AddSinkTaskCommandMsg");
    }
}

module.exports = {AddSinkTaskCommandMsg: AddSinkTaskCommandMsg};
